pkg/virt-handler/vm.go

```diff
@@ -2458,6 +2463,14 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 			if pvcutils.IsPreallocated(pvc.ObjectMeta.Annotations) {
 				preallocatedVolumes = append(preallocatedVolumes, v.Name)
 			}
+			expectedDiskSize, ok := pvcutils.ExpectedDiskSize(d.clientset, pvc)
+			if ok {
+				currentSize, ok := d.diskSizes[v.Name]
+				if ok && currentSize != expectedDiskSize {
+					d.changedDisks = append(d.changedDisks, v.Name)
+				}
+				d.diskSizes[v.Name] = expectedDiskSize
```

Comment:
this isn't thread safe

Reply:
can you explain why this needs to be thread safe?

Reply:
I reduced the scope of changedDisks --- hopefully that's what you pointed out.
diskSizes should always yield the same result, so there's less risk with multiple things editing it.